tests/unit/output/test_local.py

```diff
@@ -46,10 +47,15 @@ def test_str_on_absolute_path(dvc):
     stage = Stage(dvc)
 
     path = os.path.abspath(os.path.join("path", "to", "file"))
-    output = OutputLOCAL(stage, path, cache=False)
+    output = OutputBase(stage, path, cache=False)
```

Comment:
Why are we suddenly using OutputBase now here?

Reply:
I see, because `path` is is repo. Ok, but no need to use OutputBase, just adjust the abs path case to use an abs path from outside of repo, that is all.

Reply:
No I think we need to test both to ensure `Base` doesn't do anything clever while `LOCAL` does

Reply:
@casperdcl But why does it matter? OutputLOCAL will call `OutputBase.__init__` anyways.

Reply:
yes but I'm testing that `OutputBase` (which other outputs inherit from) doesn't alter the path, while `OutputLOCAL` does. My previous implementation would've thus failed this test. Any future work messing with `OutputBase` may also trigger this test to fail.

Reply:
@casperdcl I think the test could be improved. Right now it is really obscure how you test OutputBase here. Let's adjust the existing test to pass and then let's add a new one to test that in-repo and out-of-repo paths are resolved properly. That way it will be much cleaner.

Reply:
I split the test now - let me know if you'd rather I remove the second one